config/parse.go

```diff
@@ -0,0 +1,197 @@
+// config package provides set of utilities for configuration parsing
+package config
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/caddy/caddyfile"
+)
+
+type CfgTreeNode struct {
```

Comment:
Since this is the `config` package, no need to repeat ourselves with `Cfg` here.

Also having both "tree" and "node" is a little weird. Maybe `Node` would be enough?

Reply:
It was initially placed in the maddy package. Forgot to rename after move. Will rename to `Node`.